src/com/MPIDirectCommunication.hpp

```diff
@@ -36,24 +36,28 @@ class MPIDirectCommunication : public MPICommunication
   virtual size_t getRemoteCommunicatorSize() override;
 
   /// See precice::com::Communication::acceptConnection().
-  virtual void acceptConnection(std::string const &nameAcceptor,
-                                std::string const &nameRequester) override;
+  virtual void acceptConnection(std::string const &acceptorName,
+                                std::string const &requesterName,
+                                int                acceptorRank) override;
 
-  virtual void acceptConnectionAsServer(std::string const &nameAcceptor,
-                                        std::string const &nameRequester,
+  virtual void acceptConnectionAsServer(std::string const &acceptorName,
+                                        std::string const &requesterName,
+                                        int                acceptorRank,
                                         int                requesterCommunicatorSize) override
   {
     ERROR("Not implemented!");
```

Comment:
This should be assertions instead

Reply:
Ok, changed. Also changed one more `ERROR("Not implemented!")`. The one in `ScaleByAreaAction` I left this way, because it's code that can be triggered by the user in a wrong way. Changed in https://github.com/precice/precice/commit/9b394a671f3c4ab48f39be8ab705dd34a038b6bc